docs/proposals.rst

```diff
@@ -0,0 +1,112 @@
+.. proposals:
+
+Proposals
+=========
+
+This section should contain design and implementation proposals for Tarmak.
+Once the proposal has been implemented it should be moved to the regular user
+and/or developer documentation.
+
+
+Implement a tarmak terraform provider
+-------------------------------------
+
+Motivations
+***********
+
+Right now the terraform code for the AWS provider consists for multiple
+separate stacks (state, network, tools, vault, kubernetes). Main reason for the
+the different stacks was to enable operations in between parts of the resource
+spin up. These include (list might not be complete):
```

Comment:
Can we add an overview section? i.e. what is the actual goal of the 'terraform provider'?

Reply:
I'm actually not 100% sure what the tarmak terraform provider does. It seems like it's a replacement for the terraform defined for AWS, but I don't see how that can be as we ultimately do still need cloud-specific terraform code.

Reply:
Perhaps provide an overview of how provisioning works now, and how it will work afterwards? It may also help to have a step-by-step break down of these stages.

Reply:
Have a look through an existing design proposal in the Kubernetes repo (e.g. https://github.com/kubernetes/community/blob/master/contributors/design-proposals/cli/apply_refactor.md) and perhaps based it around that?

* Background - provide some context to the situation now (e.g. currently, each provider implements xyz, this is difficult to scale out to more clouds by xyz)

* Objective - what are we going to do to make the situation described above better? (high level design overview)

* Changes - detail the actual changes that will need to be made here. If needs be, use (annotated) code-snippets to illustrate the required changes. This *sort of* specs out the work required, but obviously is subject to change once implementation starts (if it hasn't already)

* Notable items - here we can call out things that are particularly bad, good, difficult or whatever else. Anything 'notable'.

* Out of scope - may be worth including this too, as it explicitly puts a 'bound' on the scope of the work. This may not be applicable all the time, but it's useful to include if there's any ambiguity as to whether something should or shouldn't be done.